plugins/content/document/lib/operation_engines/KOperationEngineDocument.php

```diff
@@ -56,10 +64,29 @@ protected static function jsonFormat($arr, $basePath)
 				$docMetadata[$key] = json_decode(json_encode($str), true);
 			}
 			kFile::setFileContent($basePath . self::DOC_METADATA_JSON_NAME, json_encode($docMetadata));
+
+		}
+		catch (Exception $e)
+		{
+			KalturaLog::warning("Fail to create json file: " . $e->getMessage());
+		}
+	}
+
+	protected static function encryptFileName($basePath, $fileName, $key) {
+		if (!$key )
+		{
+			return $fileName;
+		}
+		try {
+			$pathinfo = pathinfo ($fileName);
+			$newName = rtrim(base64_encode(hash_hmac('sha256', $pathinfo['filename'], $key)), "=") . '.' . $pathinfo['extension'];
```

Comment:
for shorter string maybe it's better to do 
rtrim(base64_encode(hash_hmac('sha256', 'docMetadata', "", true)), "=");

(pass true at the last character)